### PR TITLE
Introduce phasers to TR2 (part 4)

### PR DESCRIPTION
**Pull request by rr-**
#### Checklist

- [x] I have read the [coding conventions](https://github.com/LostArtefacts/TR1X/blob/master/CONTRIBUTING.md#coding-conventions)
- [x] I have added a changelog entry about what my pull request accomplishes, or it is an internal change

#### Description

Continuing from #2126, #2139 and #2140, this PR converts the inventory code to use the phaser pattern.

It also does some cleanups to the inventory ring API:

- Item management was moved to `game/inventory.c`
- Backpack module was abandoned and merged to `game/inventory.c`
- Inventory ring was entirely relocated to `game/inventory_ring/*.c`
- Variables related to the inventory got moved to `game/inventory_ring/vars.c`
- Types related to the inventory got moved to `game/inventory_ring/types.c`
- `RING_INFO` got renamed to `INV_RING`, `IMOTION_INFO` to `INV_RING_MOTION`.

While `INV_RING` now contains more state, there are still _a lot_ of global variables that I didn't dare to refactor at this time.
Most of the changes omitted TR1, which has gone too far with the phaser and need to be redone separately, once we want to combine the inventory code via libtrx.

### CODE REVIEW

**Comment by aredfan** [comment]
TR1X looks good.

- [x] In TR2X, the health bar has an odd colour when starting Lara's Home, New Game or Load Game via the main menu. It also happens when selecting a level via the console e.g. `/play 16`.

**Comment by lahm86** [comment]
In TR2, if you have music playing and go into the inventory then back to the game, the music doesn't resume.

Edit: this is on develop too, seems to have been since https://github.com/LostArtefacts/TRX/commit/268cbb878cbda5467e409b5679b50a7743d93263
This is logged after the inventory unpauses the music:
`libtrx/engine/audio_stream.c 600 Audio_Stream_Mix Error reading from sdl.stream: Parameter 'stream' is invalid`

File: src/tr2/game/inventory_ring/control.c [hunk 1]
```diff
@@ -0,0 +1,1062 @@
...
+    M_RemoveAllText();
+    if (ring->list != NULL) {
+        INV_ITEM *const inv_item = ring->list[ring->current_object];
+        if (inv_item != NULL) {
+            Option_Shutdown(inv_item);
+        }
+    }
+    Memory_Free(ring);
+    Output_UnloadBackground();
+    Music_Stop();
```
Comment (lahm86) [inline src/tr2/game/inventory_ring/control.c]: Changing this to the following resolves the music issue, not sure if you want to tackle it on develop or just fix it here as it's unreleased.
```
if (ring->mode == INV_TITLE_MODE) {
    Music_Stop();
}
```

**Comment by rr-** [comment]
@aredfan @lahm86 fixed both issues.

File: src/tr2/game/inventory_ring/control.c (hunk 1 above)
Comment (rr-) [reply src/tr2/game/inventory_ring/control.c]: Thanks, I've addressed it here.

**Comment by aredfan** [comment]
> @aredfan @lahm86 fixed both issues.

Thank you, the previous issues are resolved. I found a couple of new issues.

- [x] Use `/play 0` on the main menu, and when Lara finishes speaking, the main menu music starts playing. This issue also happens if `/play 2` or `/play 3` is used on the main menu.
- [x] Use `/demo` on the main menu, and the demo mode is shown briefly before getting skipped.

**Comment by rr-** [comment]
@aredfan fixed

**Review by aredfan — APPROVED** [review]
LGTM

**Review by lahm86 — APPROVED** [review]

**Pull request merged**